New confirmations

ShubhankarKG opened the pull request:
Fixes #1396 
Fixes #1270 
Fixes #798
Ref #1330 
Ref #1360 

Changes:-

1. Introduced three new flags for visibility filters :- `--public`, `--private`, `--internal` which need to be passed. Failure to do so invokes an interactive session.

2. Introduces new flag `-y` that passes true for all Confirm statements. `repo create` has two confirm statements :- **Confirm submit?** and **Create local directory**. Both shall be true when passed. 

3.`gh repo create` on passing no arguments shall open up the interactive session as noted.

ShubhankarKG committed:
Add confirmation for risky step for repo create

ShubhankarKG committed:
1. Add three visibility tags - public, private, internal, one of which meeds to be passed.
2. Interactive repo create when no parameters are passed, i.e `repo create`.
3. Fix tests.

TODO : write a repo in the form of user/repository format when no user is supplied.

ShubhankarKG committed:
Permit -y flag to create local directory as well

ShubhankarKG committed:
Merge branch 'trunk' into newConfirmations

ShubhankarKG committed:
Fix tests failing after merging trunk

ShubhankarKG commented:
Requesting to have a look at this :pray:

ShubhankarKG commented:
Requesting to have a look at this :pray:

ampinsk reviewed (changes requested):
Hi, thanks so much for this and apologies for the delay. I reviewed quickly from an experience perspective and had a couple of comments. I'll leave a more thorough review to @mislav or @vilmibm. thanks again!

1. In this prompt, can the options be in sentence case instead of capitalized? 

```
? Visibility  [Use arrows to move, type to filter]
> PUBLIC
  PRIVATE
  INTERNAL
```

Should be 

```
? Visibility  [Use arrows to move, type to filter]
> Public
  Private
  Internal
```
---

2. In the final confirmation, the default choice should be `yes`, not `no`, and the repository name should have single quotes around it.

`This will create name in your current directory. Continue?  (y/N)` should be `This will create 'name' in your current directory. Continue?  (Y/n)`

ShubhankarKG committed:
Merge branch 'trunk' of https://github.com/cli/cli into newConfirmations

ShubhankarKG committed:
Refactor changes

StefanXhunga commented:
Thank you for your help!

vilmibm commented on pkg/cmd/repo/create/create.go:
```diff
@@ -126,67 +180,185 @@ func createRun(opts *CreateOptions) error {
+		Name: "repoVisibility",
+		Prompt: &survey.Select{
+			Message: "Visibility",
+			Options: []string{"PUBLIC", "PRIVATE", "INTERNAL"},
```
should also become Public, Private, Internal

Jeffkeen7 replied:
Agreed

vilmibm reviewed (changes requested):
This is looking good, thank you! Just one minor change, please.

ShubhankarKG committed:
Change visibility strings to Sentence case

ShubhankarKG committed:
doSetup wasn't getting called in cases where required.

The problem was that opts.confirmSubmit was mutated before reaching doSetup. This commit creates a copy of the initial confirmSubmit value. So the doSetup receives the initial data passed from the command, not the mutated one.

StefanXhunga commented:
Hi All,
thank you for your interest to help me!

vilmibm commented:
Thank you for your work and patience!

tonyandbrenda2019 commented:
this is very different
